fw/http.c

```diff
@@ -3029,14 +3029,49 @@ tfw_h1_rewrite_purge_to_get(struct sk_buff *head)
 	return -ENOMEM;
 }
 
+/**
+ * Removes n bytes from the beginning of the message, iterating over skb chain if needed.
+ * @return zero on success and error code on error.
+ *
+ * TODO: to be replaced with future version of ss_skb_chop_head_tail() over all skbs of the message
+ */
+static int
+tfw_h1_chop_leading_crlf(struct sk_buff **head, unsigned n)
+{
+	struct sk_buff *skb = *head;
+	while (n) {
+		if (likely(skb->len > n))
+			return ss_skb_chop_head_tail(skb, skb, n, 0);
+		if (unlikely(skb->next == skb))
```

Comment:
It seems that you remove the skb by hands instead of using `ss_skb_unlink()` only because of this check: to return error if `n` is larger than the whole list data size. Please comment it - I spent quite some time to understand why don't you use `ss_skb_unlink()`.

Reply:
It seems, this check do not influence, because it precede.
I make this check because I do not know how the subsequent request processing functions will react on emty request.

Probably it was more simple for me to code the action by myself rather than make a special search for special function like `ss_skb_unlink().` :)
However, looking into it, I see it perform some unnessessary actions.
1) It checks for the `head`. I know apriori that the buf is a `head`.
2) It clears `prev`, `next`. I know apriori that the buf is to be freed and do not worry about it fields.